synapse/third_party_rules/access_rules.py

```diff
@@ -561,20 +559,29 @@ async def _freeze_room_if_last_admin_is_leaving(
         #     "kick": 100,
         #     "invite": 100
         # }
+        new_content = {}
         for key, value in power_level_content.items():
             # Do not change "users_default", as that key specifies the default power
             # level of new users
             if isinstance(value, int) and key != "users_default":
-                power_level_content[key] = 100
-        power_level_content["events"] = {}
+                value = 100
+            new_content[key] = value
+
+        # Clear out any special-case event power levels
+        new_content["events"] = {}
+
+        # Ensure state_default and events_default keys exist and are 100
+        # Else a lower PL user could potentially send state events that
+        # aren't explicitly mentioned elsewhere in the power level dict
+        new_content["state_default"] = 100
+        new_content["events_default"] = 100
```

Comment:
we might still have issues with missing `ban`, `kick`, etc keys, each of which default to 50?

Reply:
where is defined this default to 50 for missing `ban`, `kick` and others?

Reply:
It's written in the spec: https://matrix.org/docs/spec/client_server/r0.6.1#m-room-power-levels

![image](https://user-images.githubusercontent.com/1342360/95851851-c17f0480-0d4a-11eb-831c-a0628b7cc6b4.png)

Reply:
We're going to go with setting these to 100 if they're missing for now as we set them to 100 above if they're present anyways.